src/msg/Messenger.cc

```diff
@@ -22,10 +22,12 @@ Messenger *Messenger::create(CephContext *cct, const string &type,
 			     entity_name_t name, string lname,
 			     uint64_t nonce, uint64_t features)
 {
-  int r = -1;
-  srand(time(NULL));
-  if (type == "random")
+  static int r = -1;
```

Comment:
Thanks for the comment.  I think std_once is overkill here. 
std::once_flag / std::call_once() are when thread safety is needed.  Our code is only being called from one thread at initialization - usually from 'main() function.  Even if someone will call it it in the future from multiple threads in parallel with a race condition - still it will not affect safety of the program, but it will at most affect the selected random value in a case a race condition will be resulted in multiple calls to srand.  BTW, We can consider this future race as an additional random factor :-)

Also, please notice, that the original code already used srand() and rand().  These functions work by keeping hidden state internally with the 'seed' for next call .  In case you really want comprehensive support in multi-threaded environment then we should call rand_r() instead of srand() & rand().